### ISSUE TITLE
Current preferred tensorflow python api interface

### ISSUE THREAD

**Issue by mapa17** [tensorflow/models#4715]
Hello I am very new to TF and a bit overwhelmed by the different python API interfaces (using Estimators, TF Eager Execution, Keras, ...?). It seems to me that each new mayor TF release brings another interface. So my questions

What is the current (July 2018, TF v1.8) preferred *way* to
* write CNN for image classification
* write your own input data function (ETL data)
* being able to define the network architecture (layers, activation functions, optimizer, loss function)
* store and load trained models

What I am not interested in the moment is to be able to write my own layers, optimizers of loss functions.

Thank you very much!

PS: I am not able to at the moment use stack overflow; otherwise i would have put my question there.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by mapa17** [comment]
What is the top-level directory of the model you are using: None
Have I written custom code: No
OS Platform and Distribution: MacOs
TensorFlow installed from: conda
TensorFlow version: 1.6
Bazel version: no clue
CUDA/cuDNN version: no clue
GPU model and memory: no clue
Exact command to reproduce: None

**Comment by mapa17** [comment]
Sorry, but this is really ridiculous. I was in the process of reading the documentation on the TF homepage, when 1.9 was released ... and what a surprise; the complete documentation was restructured; going away from the 1.8 Getting Started linear path to a pick and choose modular documentation.

For all the confused beginners like me out there, my recommendation:

**Learn how to use Estimators with TF**

By reading:
* [Basics about premade estimators] (https://www.tensorflow.org/guide/premade_estimators)
* [About checkpoints and the benefits of estimators] (https://www.tensorflow.org/guide/checkpoints)
* [What input functions are and how to get data into your estimator] (https://www.tensorflow.org/versions/r1.3/get_started/input_fn)
* [How to describe your features and do a bit of feature engineering](https://www.tensorflow.org/guide/feature_columns)
* [Learn more about how to handler your input data] (https://www.tensorflow.org/guide/datasets_for_estimators)
* [Finally how to write your own estimator] (https://www.tensorflow.org/guide/custom_estimators)

To my understanding you get the full power of TF with estimators, and to your liking can even use Keras to develop the model part, and afterwards [convert it into an estimator] (https://www.tensorflow.org/guide/keras#distribution)

**Comment by karmel** [comment]
@lamberta This issue has been closed, but JFYI

**Comment by lamberta** [comment]
Hi Pasieka,
Thanks for the feedback, I've reordered the estimator guides based on your suggestion: https://github.com/tensorflow/tensorflow/commit/34c427aecd0fcc23816bc04399c6d40022ca4480
The TensorFlow platform supports multiple APIs, we're working to clarify why you would pick one over another.
